src/audio/smart_amp/smart_amp_maxim_dsm.c

```diff
@@ -60,6 +60,162 @@ static int maxim_dsm_init(struct smart_amp_mod_struct_t *hspk, struct comp_dev *
 	return (int)retcode;
 }
 
+static int maxim_dsm_get_all_param(struct smart_amp_mod_struct_t *hspk,
+				   struct comp_dev *dev)
+{
+	struct smart_amp_caldata *caldata = &hspk->param.caldata;
+	int32_t *db = (int32_t *)caldata->data;
+	enum DSM_API_MESSAGE retcode;
+	int cmdblock[1 + DSM_DEFAULT_NUM_CHANNEL];
+	int num_param = hspk->param.max_param;
+	int x;
+
+	for (x = 0 ; x < num_param ;  x++) {
+		cmdblock[0] = DSM_SET_CMD_ID(x);
+		retcode = dsm_api_get_params(hspk->dsmhandle, 1, (void *)cmdblock);
+		if (retcode != DSM_API_OK) {
+			/* set zero if the parameter is not readable */
+			cmdblock[1] = 0;
+			cmdblock[2] = 0;
+		}
+
+		db[x * DSM_PARAM_MAX + DSM_PARAM_ID] = 0x01000000 | x;
+		db[x * DSM_PARAM_MAX + DSM_PARAM_VALUE] = cmdblock[1];
+		db[(x + num_param) * DSM_PARAM_MAX + DSM_PARAM_ID] = 0x02000000 | x;
+		db[(x + num_param) * DSM_PARAM_MAX + DSM_PARAM_VALUE] = cmdblock[2];
+	}
+
+	return 0;
+}
+
+static int maxim_dsm_get_volatile_param(struct smart_amp_mod_struct_t *hspk,
+					struct comp_dev *dev)
+{
+	struct smart_amp_caldata *caldata = &hspk->param.caldata;
+	int32_t *db = (int32_t *)caldata->data;
+	enum DSM_API_MESSAGE retcode;
+	int cmdblock[1 + DSM_DEFAULT_NUM_CHANNEL];
+	int num_param = hspk->param.max_param;
+	int x;
+
+	/* Update all volatile parameter values */
+	for (x = DSM_API_ADAPTIVE_PARAM_START ; x < DSM_API_ADAPTIVE_PARAM_END ;  x++) {
+		cmdblock[0] = DSM_SET_CMD_ID(x);
+		retcode = dsm_api_get_params(hspk->dsmhandle, 1, (void *)cmdblock);
+		if (retcode != DSM_API_OK)
+			return -EINVAL;
+
+		db[x * DSM_PARAM_MAX + DSM_PARAM_ID] = 0x01000000 | x;
+		db[x * DSM_PARAM_MAX + DSM_PARAM_VALUE] = cmdblock[1];
+		db[(x + num_param) * DSM_PARAM_MAX + DSM_PARAM_ID] = 0x02000000 | x;
+		db[(x + num_param) * DSM_PARAM_MAX + DSM_PARAM_VALUE] = cmdblock[2];
+	}
+
+	return 0;
```

Comment:
ditto for comments and magic numbers in this func too.

Reply:
Thanks. Shall add more comments.